homeassistant/components/linky/sensor.py

```diff
@@ -29,29 +50,64 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     username = config[CONF_USERNAME]
     password = config[CONF_PASSWORD]
     timeout = config[CONF_TIMEOUT]
+    monitored_conditions = config[CONF_MONITORED_CONDITIONS]
+
+    LinkyAccount(hass, add_entities, username, password, timeout,
+                 monitored_conditions)
 
-    from pylinky.client import LinkyClient, PyLinkyError
-    client = LinkyClient(username, password, None, timeout)
-    try:
-        client.login()
-        client.fetch_data()
-    except PyLinkyError as exp:
-        _LOGGER.error(exp)
-        client.close_session()
-        return
 
-    devices = [LinkySensor('Linky', client)]
-    add_entities(devices, True)
+class LinkyAccount:
+    """Device manager for demo geolocation events."""
+
+    def __init__(self, hass, add_entities, username, password, timeout,
+                 monitored_conditions):
+        """Initialise the demo geolocation event manager."""
+        self.hass = hass
+        self._client = LinkyClient(username, password, None, timeout)
+
+        try:
+            self._client.login()
+            self._client.fetch_data()
+        except PyLinkyError as exp:
+            _LOGGER.error(exp)
+            self._client.close_session()
+            return
+
+        self.__devices = []
+        for condition in monitored_conditions:
+            self.__devices.append(
+                LinkySensor(SENSORS[condition][SENSORS_INDEX_LABEL],
+                            self._client,
+                            SENSORS[condition][SENSORS_INDEX_SCALE],
+                            SENSORS[condition][SENSORS_INDEX_WHEN]))
+
+        add_entities(self.__devices, True)
```

Comment:
The class `LinkyAccount` should handle the client part and the `setup_platform` the setup of the entities.